### PR TITLE
Do not create end product establishments for ineligible issues

### PR DESCRIPTION
**Pull request by leikkisa**
connects #9249

### CODE REVIEW

File: app/models/claim_review.rb [hunk 1]
```diff
@@ -199,6 +199,8 @@ def intake_processed_by
   end
 
   def end_product_establishment_for_issue(issue)
+    return unless issue.eligible?
```
Comment (pkarman) [inline app/models/claim_review.rb]: I wonder if this is the right place for this?

if you put the filter for `.eligible?` inside `create_issues!` then you could avoid calling this at all.

Comment (pkarman) [reply app/models/claim_review.rb]: e.g.
```diff
@@ -56,7 +56,7 @@ class ClaimReview < DecisionReview
         issue.update!(benefit_type: benefit_type, veteran_participant_id: veteran.participant_id)
       else
         issue.update!(
-          end_product_establishment: end_product_establishment_for_issue(issue),
+          end_product_establishment: issue.eligible? ? end_product_establishment_for_issue(issue) : nil,
           benefit_type: benefit_type,
           veteran_participant_id: veteran.participant_id
         )
```

I think I would find that more explicit.

Comment (pkarman) [reply app/models/claim_review.rb]: alternately:
```diff
@@ -52,7 +52,7 @@ class ClaimReview < DecisionReview
   # Create that end product establishment if it doesn't exist.
   def create_issues!(new_issues)
     new_issues.each do |issue|
-      if processed_in_caseflow?
+      if processed_in_caseflow? || !issue.eligible?
         issue.update!(benefit_type: benefit_type, veteran_participant_id: veteran.participant_id)
       else
         issue.update!(
```

File: spec/models/claim_review_spec.rb [hunk 2]
```diff
@@ -325,6 +328,17 @@ def random_ref_id
     context "when there's more than one issue" do
       let(:issues) { [rating_request_issue, non_rating_request_issue] }
 
+      context "when they're all ineligible" do
```
Comment (pkarman) [inline spec/models/claim_review_spec.rb]: good test

File: app/models/claim_review.rb (hunk 1 above)
Comment (leikkisa) [reply app/models/claim_review.rb]: I like that!

**Pull request merged**